bulk-update/document-manager/document-manager.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable max-len */
+import { fetch } from '@adobe/fetch';
+import { mdast2docx } from '@adobe/helix-md2docx';
+import parseMarkdown from '@adobe/helix-html-pipeline/src/steps/parse-markdown.js';
+
+import fs from 'fs';
+
+const delay = (milliseconds) => new Promise((resolve) => { setTimeout(resolve, milliseconds); });
+
+/**
+ * Format the path to the actual url/file.
+ * Making sure URLs ending in a slash are an index file.
+ *
+ * @param {string} entry - The index entry or URL
+ * @returns {string} - The markdown path
+ */
+export function entryToPath(entry) {
+  return `${entry.split(/\?|#/)[0].replace(/\/$/, '/index')}`;
+}
+
+/**
+ * Fetches a markdown file from a given URL and optionally saves it locally.
+ *
+ * If the `url` option is provided, the function fetches the markdown from the given URL.
+ * If the `mdDir` option is provided, it saves the fetched markdown to the specified local path.
+ *
+ * @param {string} url - The URL to fetch the markdown file from.
+ * @param {string} mdDir - The local path to save the fetched markdown to.
+ * @returns {string} - The markdown content as a string or null if not found or an error occurred.
+ */
+async function getMarkdown(url, reporter) {
+  try {
+    await delay(500); // Wait 500ms to avoid rate limiting
```

Comment:
Is 500ms an official time from microsoft or is that an estimate? Ideally we'd want it to be as short as possible without hitting the rate limit

Reply:
This is a good point, this may not be hitting the sharepoint api, even more so for hlx.live, so this could be potentially set to a lower value. 500 ms has been used as a safe value for other request, how about we setting 500 as a default parameter and then it can be overridden by the config.

Reply:
Thanks for the suggestion, this helps speed up the tests.